api/client/network.go

```diff
@@ -222,7 +224,7 @@ func consolidateIpam(subnets, ranges, gateways []string, auxaddrs map[string]str
 				return nil, fmt.Errorf("multiple overlapping subnet configuration is not supported")
 			}
 		}
-		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}}
+		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}, Options: options}
```

Comment:
I understand your point. But from an UX point of view, having a subnet specific option will not work. also, from a functionality standpoint, I dont think it is appropriate to have a subnet specific option. But as you said ipam API takes in per-subnet options because IPAM drivers have no concept of a `network`. LibNetwork moderates the network  requests and makes subnet specific request to the IPAM driver. I guess this falls into the same category.
IMO, `-ipam-option` must be made NetworkCreate specific & libnetwork can internally use it to send it (for each subnet) to the IPAM driver. WDYT ?
ping @aboch